system-addon/test/unit/content-src/lib/detect-user-session-start.test.js

```diff
@@ -0,0 +1,56 @@
+const DetectUserSessionStart = require("content-src/lib/detect-user-session-start");
+const {actionTypes: at} = require("common/Actions.jsm");
+
+describe("detectUserSessionStart", () => {
+  describe("#init", () => {
+    it("should call ._sendEvent immediately if the document is visible", () => {
+      const mockDocument = {visibilityState: "visible"};
+      const instance = new DetectUserSessionStart({document: mockDocument});
+      sinon.stub(instance, "_sendEvent");
+
+      instance.init();
+
+      assert.calledOnce(instance._sendEvent);
```

Comment:
personally I think individual units/functions should be tested separately whether they are public or private since isolating breakages during refactors is easier, and changing private apis does cause (private) breakages anyway